byzcoin/bcadmin/lib/utils.go

```diff
@@ -142,6 +143,50 @@ func CombinationAnds(list []string, m int) []string {
 	return list
 }
 
+// WaitPropagation contacts all nodes in the cl.Roster until they all
+// have the same latest block. If there is an error when calling
+// `GetProof`, the error will be ignored. This helps when waiting
+// for the propagation, but only a subset of the nodes are actually
+// participating in the consensus.
+func WaitPropagation(c *cli.Context, cl *byzcoin.Client) error {
+	if !c.GlobalBool("wait") {
+		return nil
+	}
+
+	var sb skipchain.SkipBlock
+	sb.SkipBlockFix = &skipchain.SkipBlockFix{}
+searchLatest:
+	for {
+		log.Lvl2("Starting search")
+		for node := range cl.Roster.List {
+			log.Lvl2("Searching node", node)
+			if err := cl.UseNode(node); err != nil {
+				return err
+			}
+			_, err := cl.GetProof(make([]byte, 32))
+			if err != nil {
+				log.Warn("error while searching for node - ignoring")
+				//continue searchLatest
+			}
+			if cl.Latest.Index > sb.Index {
+				sb = *cl.Latest
+				log.Lvl2("Found new block:", sb.Index)
+				cc, err := cl.GetChainConfig()
+				if err != nil {
+					return err
+				}
+				cl.Roster = cc.Roster
+				continue searchLatest
+			} else if cl.Latest.Index < sb.Index {
+				log.Lvlf2("Node %d returned earlier block: %d", node, cl.Latest.Index)
+				continue searchLatest
```

Comment:
what would happen if one of the node always returns an earlier block? will this function wait infinitely? i think the outer for-loop should have some kind of limit

Reply:
OK, added an outer for-loop and a wait for every subsequent query, to avoid pinging the nodes too frequently. With javascript I would've added default parameters, but that doesn't exist in go, so I hard-coded them ;)